### PR TITLE
Fix gaussian_nll_loss eps docstring: it clamps var, not adds

### PR DESCRIPTION
**Pull request by vineethsaivs**
### Summary

`torch.nn.functional.gaussian_nll_loss` documents `eps` as:

> eps (float, optional): Value added to var, for stability. Default: 1e-6.

but the implementation clamps the variance to a floor rather than adding `eps`:

```python
var.clamp_(min=eps)   # i.e. max(var, eps)
```

So the loss tracks `max(var, eps)`, not `var + eps`. Quick check on 2.11 (input == target == 0, so `loss == 0.5 * log(effective_var)`):

```
v=4.0,  eps=1.0 -> loss=0.693147  (== 0.5*log(max(4, 1))=0.693147, not 0.5*log(4+1)=0.804719)
v=0.25, eps=1.0 -> loss=0.000000  (== 0.5*log(max(0.25, 1))=0.0,     not 0.5*log(0.25+1)=0.111572)
```

The sibling module `torch.nn.GaussianNLLLoss` already documents this correctly ("value used to clamp `var`"). This updates the functional docstring to match both the code and the module docstring.

### Change

```
-        eps (float, optional): Value added to var, for stability. Default: 1e-6.
+        eps (float, optional): Value used to clamp ``var`` to a minimum, for stability. Default: 1e-6.
```

Documentation only; no functional change.

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
thanks for the catch! also fix in torch/nn/modules/loss.py please?

**Comment by vineethsaivs** [comment]
Thanks for taking a look!

I checked `torch/nn/modules/loss.py` and `GaussianNLLLoss` already documents this correctly, which is actually where I took the corrected wording from:

```
eps (float, optional): value used to clamp ``var`` (see note below), for
    stability. Default: 1e-6.
```

and its math block already uses `\text{max}\left(\text{var}, \ \text{eps}\right)` rather than an addition.

A repo-wide search for the old phrasing turns up only the single line in `torch/nn/functional.py` that this PR changes, so `functional.py` was the last place still saying "added to var". Happy to update anything else if you had a different spot in mind.

**Comment by janeyx99** [comment]
Ah I meant as a minimum, to have the same wording, but you're right it isn't really clearer.

**Review by janeyx99 — APPROVED** [review]
